crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -91,6 +91,10 @@ pub(super) struct SlidingSyncInner {
     /// The storage key to keep this cache at and load it from.
     storage_key: String,
 
+    /// Should this sliding sync instance try to restore its sync position
+    /// from the database?
+    share_pos: bool,
```

Comment:
Again, behind `cfg(feature = "e2e-encryption)`?

Reply:
... but this one would make the code really harder to read (you'll see that to accomodate the above request, I'm already adding lots of cfg-guarded statements). i'm slightly in favor of keeping it to `false` for non-`e2e-encryption` builds. Thoughts?

Reply:
OK. Let's add a comment to explain this maybe?

Reply:
Sounds good!